docs/technical_specs/model_libraries/generic/unit_models/valve.rst

```diff
@@ -0,0 +1,222 @@
+  cv = fin/math.sqrt(pin - pout)/0.5
+  # set inlet
+  m.fs.valve.inlet.enth_mol[0].fix(hin)
+  m.fs.valve.inlet.flow_mol[0].fix(fin)
+  m.fs.valve.inlet.pressure[0].fix(pin)
+  m.fs.valve.Cv.fix(cv)
+  m.fs.valve.valve_opening.fix(0.5)
+
+  iscale.calculate_scaling_factors(m)
+  m.fs.valve.initialize(outlvl=1)
+
+  solver = pyo.SolverFactory("ipopt")
+  solver.options = {"nlp_scaling_method": "user-scaling"}
+  solver(m, tee=True)
+
+
+Variables
+---------
+
+======================= ======================== =========== =======================================================
+Variable                Symbol                   Index Sets  Doc
+======================= ======================== =========== =======================================================
+``Cv``                  :math:`C_v`              None        Valve coefficient
+``valve_opening``       :math:`x`                time        The fraction that the valve is open from 0 to 1
+======================= ======================== =========== =======================================================
+
+The ``Cv`` variable is highly recommended but can be omitted in custom pressure-flow relations.
+
+Expressions
+-----------
+
+=========================== ========================= =========== ===========================================================================
+Expression                  Symbol                    Index Sets  Doc
+=========================== ========================= =========== ===========================================================================
+``valve_function``          :math:`f(x)`              time        This is a valve function that describes how the fraction open affects flow.
+=========================== ========================= =========== ===========================================================================
+
+Built-in Valve Functions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Standard valve functions can be specified by providing a ``ValveFunctionType``
+enumerate type to the ``valve_function_callback`` argument.  Standard function
+are given below.
+
+``ValveFunctionType.linear``
+
+.. math::
+
+  f(x) = x
+
+``ValveFunctionType.quick_opening``
+
+.. math::
+
+  f(x) = \sqrt{x}
+
+``ValveFunctionType.equal_percentage``
+
+.. math::
+
+  f(x) = \alpha^{x - 1}
+
+For the equal-percentage valve function and additional variable ``alpha`` is defined
+which by default is fixed and set to 100.
+
+Custom Valve Functions
+~~~~~~~~~~~~~~~~~~~~~~
+
+In general, the valve opening should be restricted to range from 0 to 1.  The valve
+function should be a named expression attached to the valve model called
+``valve_function`` which takes the valve opening and computes a value that goes
+from approximately zero when valve opening is 0 to 1 when the valve opening is one.
+The valve function can have parameters as needed, so custom valve functions are
+defined in a callback so any additional parameters can be added in the callback.
```

Comment:
I suggest rephrasing this sentence to: "Custom valve functions are defined using a callback function." 

The point about being able to include additional parameters is made in the next paragraph.